src/App.tsx

```diff
@@ -1,24 +1,231 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  createTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 10999;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.ALL);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const getAllTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorMessage('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timeout = setTimeout(() => {
+        setErrorMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+
+    return () => {};
+  }, [errorMessage]);
+
+  const hasActiveTodo = todos?.find((todo) => todo.completed === false);
```

Comment:
```suggestion
  const hasActiveTodo = !!todos?.some((todo) => !todo.completed);
```
looks like we can store in this var a boolean value